packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx

```diff
@@ -307,6 +320,28 @@ export default function CreateThread({
         } catch (e) {
           await tryFetchAddressFromTwitterHandle(address);
         }
+      } else if (address) {
+        try {
+          if (program?.provider.connection) {
+            const pubKey = new anchor.web3.PublicKey(address);
+
+            const fetchPromises = [
+              tryGetName(program?.provider.connection, pubKey),
+              fetchSolanaNameServiceName(program?.provider.connection, address),
+            ];
+
+            // TODO: Fix typing on promise result
+            const result: any = await Promise.all(fetchPromises);
```

Comment:
For better readability, let's destruct the array and name the results.
```
const [name, snsName] = await Promise.all([
  ... // no need for an extra variable
];

if (name && snsName) { ... }
```

Reply:
Sounds good. let's go with `twitterName` and `snsName`.